### ISSUE TITLE
Failed to find Oxen Wallet RPC. Please make sure your anti-virus has not removed it.

### ISSUE THREAD

**Issue by agitationswapping** [loki-project/ryo-wallet-libre#332]
Hi all,

I have been using the Oxen Electron Wallet v1.8.1 without issue for many months and I haven't run it for many weeks. Yesterday I went to open the wallet and it failed part way after connecting to the daemon and during setting up the wallet. Now when I try to load the wallet, I get error in the title every time.

I can click on English to see:
Remote Host Node: public-eu.optf.ngo
Port: 22023

The error message seems to be suggesting that maybe some files are removed or inaccessible. However, from the interface, it feels like maybe I need to specify a different node.

I found someone else had this error here:
https://github.com/oxen-io/oxen-electron-gui-wallet/issues/299
It suggests maybe to update, but I am already using the latest version.

![IssueConnectingRPC](https://github.com/oxen-io/oxen-electron-gui-wallet/assets/111817552/2e72d1e6-1dfc-4bf1-b776-dd03a9383aec)

I appreciate any help or ideas! Thanks so much!

**Comment by exolocity** [comment]
Try running it as administrator, that'll fix it.  @agitationswapping 

Also make sure you have all the oxen folders and files, in windows theyre located in 3 spots;

#1: %AppData%  - "C:\Users\##YOURUSERNAME###\AppData\Roaming\Oxen Electron Wallet" 

#2: ProgramData folder - "C:\ProgramData\oxen" (this is the default location for the blockchain storage iirc, current full local download size is ~17gb)

#3: wallet location - C:\Users\YOURUSERNAME\Documents\Oxen

#4: one file installer.exe inside -> C:\Users\YOURUSERNAME\AppData\Local\oxen-electron-wallet-updater\

**Comment by agitationswapping** [comment]
Thanks for your attempt to help and sorry that I haven't been able to follow up on this sooner.

Running as administrator makes no difference. The error is the same.

%AppData% exists and appears to have lots of subfolders blob_storage, Cache, GPUCache, Local Storage, logs. logs folder is empty so no useful error report there. As a test, I renamed this entire folder. The daemon recreated a similar structure from scratch, and it didn't fix the problem.

ProgramData/oxen folder exists and only has gui folder with config.json in there. I had selected to store blockchain data in a different folder, and that folder only has a logs subdirectory. There is one file in there with an error about a promise rejection failure. That is nowhere near 17Gb. Moving this back to use ProgramData/oxen didn't fix the issue, nor did renaming the folder I originally created, nor did renaming the ProgramData/oxen folder so it would be recreated.

Documents/Oxen wallet location had been moved to a subfolder location as I was cleaning up my documents folder. (It's a pretty standard location which is really not a good idea to store data there.) Changing the directory in the Configuration, or moving the folder back, did not, however, fix the issue. And it's worth noting that Oxen had no challenge recreating this folder with an empty wallet.

The installer.exe file was also present in the directory as mentioned.

What did ultimately appear to fix the problem was renaming the installation directory of AppData\Local\Programs\Oxen Electron Wallet to a different name and then running the installer. As I mentioned, I had previously attempted to reinstall the wallet multiple times without renaming the folder and it did not work. A comparison of the two folders (the original and the renamed folder) with WinMerge shows that the sole difference is the file \resources\bin\oxen-wallet-rpc.exe. This file was missing from the version of the program which was installed on my computer.

So it seems from this, that there is an issue with the installer. When it's run it doesn't seem to detect and fix/recreate the missing oxen-wallet-rpc.exe file. I have no idea how this file got to be destroyed but I suspect that, as stated, it was an issue with anti-virus. Windows Defender has reported multiple issues and classified parts of the Oxen Electrum Wallet as Trojan:Win32/Wacatac.A!ml. I had to manually override this in order to get things working.

The bottom line is that the handling of this failure could be improved. I couldn't use this wallet for months and had to spend several hours troubleshooting. As a result, I've missed out on so many of the staking rewards in the Session Token Swap Program. Adding the location where the RPC is could be very helpful to the error message. Having the installer restore the RPC executable could be very helpful as well. Some sort of log with additional details when errors like this happen could be really useful too. I don't know that Oxen will be maintained much going forward, but hopefully errors are better handled in whatever wallet Session provides for staking.

**Comment by exolocity** [comment]
> Thanks for your attempt to help and sorry that I haven't been able to follow up on this sooner.
> 
> Running as administrator makes no difference. The error is the same.
> 
> %AppData% exists and appears to have lots of subfolders blob_storage, Cache, GPUCache, Local Storage, logs. logs folder is empty so no useful error report there. As a test, I renamed this entire folder. The daemon recreated a similar structure from scratch, and it didn't fix the problem.
> 
> ProgramData/oxen folder exists and only has gui folder with config.json in there. I had selected to store blockchain data in a different folder, and that folder only has a logs subdirectory. There is one file in there with an error about a promise rejection failure. That is nowhere near 17Gb. Moving this back to use ProgramData/oxen didn't fix the issue, nor did renaming the folder I originally created, nor did renaming the ProgramData/oxen folder so it would be recreated.
> 
> Documents/Oxen wallet location had been moved to a subfolder location as I was cleaning up my documents folder. (It's a pretty standard location which is really not a good idea to store data there.) Changing the directory in the Configuration, or moving the folder back, did not, however, fix the issue. And it's worth noting that Oxen had no challenge recreating this folder with an empty wallet.
> 
> The installer.exe file was also present in the directory as mentioned.
> 
> What did ultimately appear to fix the problem was renaming the installation directory of AppData\Local\Programs\Oxen Electron Wallet to a different name and then running the installer. As I mentioned, I had previously attempted to reinstall the wallet multiple times without renaming the folder and it did not work. A comparison of the two folders (the original and the renamed folder) with WinMerge shows that the sole difference is the file \resources\bin\oxen-wallet-rpc.exe. This file was missing from the version of the program which was installed on my computer.
> 
> So it seems from this, that there is an issue with the installer. When it's run it doesn't seem to detect and fix/recreate the missing oxen-wallet-rpc.exe file. I have no idea how this file got to be destroyed but I suspect that, as stated, it was an issue with anti-virus. Windows Defender has reported multiple issues and classified parts of the Oxen Electrum Wallet as Trojan:Win32/Wacatac.A!ml. I had to manually override this in order to get things working.
> 
> The bottom line is that the handling of this failure could be improved. I couldn't use this wallet for months and had to spend several hours troubleshooting. As a result, I've missed out on so many of the staking rewards in the Session Token Swap Program. Adding the location where the RPC is could be very helpful to the error message. Having the installer restore the RPC executable could be very helpful as well. Some sort of log with additional details when errors like this happen could be really useful too. I don't know that Oxen will be maintained much going forward, but hopefully errors are better handled in whatever wallet Session provides for staking.

### Damn this was totally my fault, i forgot to mention that the whole reason for making sure the files are all there is because windows defender deletes them!! I just checked my exclusions and I have all the folders i listed above as "exclusions" inside windows Security:

**Folder Exlusions inside Windows Security:**
`C:\ProgramData\oxen`
`C:\Users\USERNAME\AppData\Roaming\Oxen Electron Wallet`
`C:\Users\USERNAME\AppData\Roaming\oxen-electron-wallet-updater`
`C:\Users\USERNAME\AppData\Local\oxen-electron-wallet-updater`
`C:\Users\USERNAME\Documents\Oxen`

**I also have Process Exlusions inside Windows Security:**
`Oxen Electron Wallet.exe`
`oxen-wallet-rpc.exe`

**_I have these also for Monero & XMRig_**
**Folders:**
`C:\Users\USERNAME\Documents\Monero`

**Processes:**
`monerod.exe`
`monero-wallet-gui.exe`
`xmrig.exe`

My bad @agitationswapping. I'm guessing that if you check your windows security quarantine you may Oxen Files there.
